libbeat/autodiscover/providers/kubernetes/service.go

```diff
@@ -0,0 +1,185 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kubernetes
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gofrs/uuid"
+	k8s "k8s.io/client-go/kubernetes"
+
+	"github.com/elastic/beats/libbeat/autodiscover/builder"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/bus"
+	"github.com/elastic/beats/libbeat/common/kubernetes"
+	"github.com/elastic/beats/libbeat/common/safemapstr"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type service struct {
+	uuid    uuid.UUID
+	config  *Config
+	metagen kubernetes.MetaGenerator
+	logger  *logp.Logger
+	publish func(bus.Event)
+	watcher kubernetes.Watcher
+}
```

Comment:
pod.go is fairly well tested, could you add some tests for service and node too?

Reply:
added test cases for node and service.